tests/partitioninfo.spec.coffee

```diff
@@ -25,3 +30,55 @@ describe 'Partitioninfo:', ->
 				m.chai.expect(promise).to.eventually.become
 					offset: 4194304
 					size: 20971520
+
+	describe '.getPartitions()', ->
+
+		describe 'given a valid image file', ->
+
+			beforeEach ->
+				@bootRecordReadStub = m.sinon.stub(bootRecord, 'read')
+				@bootRecordReadStub.withArgs('mbr/rpi', 0).returns(Promise.resolve(rpiMBR))
+				@bootRecordReadStub.withArgs('mbr/rpi', 394264576).returns(Promise.resolve(bbbMBR))
```

Comment:
@jviotti the raspberry pi MBR has an extended partition, and so getPartitions will go read that MBR. If return the raspberry pi MBR again it gets into an infinite loop. So I used the beaglebone MBR because it doesn't have any extended partitions.